include/crpropa/Vector3.h

```diff
@@ -28,32 +28,63 @@ namespace crpropa {
 template<typename T>
 class Vector3 {
 public:
-	T x, y, z;
+	// While data is stored in this array, it is also accessible via x,y,z variables
+	// x,y,z cannot be used in constructors or assignment operators to avoid
+	// uninitialized rvalue assignment!
+	T data[3];
 
-	Vector3() :
-			x(0), y(0), z(0) {
+	T& x = data[0];
+	T& y = data[1];
+	T& z = data[2];
+
+	Vector3() : data{0., 0., 0.} {
+	}
+
+	// avoid creation of default non-conversion constructor
+	Vector3(const Vector3 &v) : data{v.data[0], v.data[1], v.data[2]} {
 	}
 
 	// Provides implicit conversion
 	template<typename U>
-	Vector3(const Vector3<U> &v) :
-			x(v.x), y(v.y), z(v.z) {
+	Vector3(const Vector3<U> &v) {
+		data[0] = v.x;
+		data[1] = v.y;
+		data[2] = v.z;
+	}
+
+	//Vector3(Vector3 &&v) noexcept {
```

Comment:
why the noexcept version is commented out?

Reply:
I started to implement a move constructor (rule-of-5 in C++11 instead of rule-of-3) - I thought this a reason for a segfault I had before - but this is not needed.

There is no change in the -builtin version here, as the issue is with [SWIG](https://github.com/swig/swig/issues/1595#issuecomment-541809053) we should probably create a test for the issue and a test for the switch version as soon as this is resolved. I will create an issue for this so that this is not forgotten.